dev/breeze/src/airflow_breeze/utils/github.py

```diff
@@ -177,3 +180,123 @@ def get_tag_date(tag: str) -> str | None:
         tag_object.committed_date if hasattr(tag_object, "committed_date") else tag_object.tagged_date
     )
     return datetime.fromtimestamp(timestamp, tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def download_artifact_from_run_id(run_id: str, output_file: str):
+    """
+    Downloads a file from GitHub Actions artifact
+
+    :param run_id: run_id of the workflow
+    :param output_file: Path where the file should be downloaded
+    """
+    import requests
+    from tqdm import tqdm
+
+    url = f"https://api.github.com/repos/apache/airflow/actions/runs/{run_id}/artifacts"
+    headers = {"Accept": "application/vnd.github.v3+json"}
+
+    session = requests.Session()
+    if os.getenv("GITHUB_TOKEN"):
+        headers["Authorization"] = f"Bearer {os.getenv  ('GITHUB_TOKEN')}"
+
+    artifact_response = requests.get(url, headers=headers)
+
+    if artifact_response.status_code != 200:
+        get_console().print(
+            "[error]Describing artifacts failed with status code %s, "
+            "you might need to provide GITHUB_TOKEN, set it as environment variable",
+            artifact_response.status_code,
+        )
+        sys.exit(1)
+
+    download_url = None
+    file_name = os.path.splitext(os.path.basename(output_file))[0]
+    for artifact in artifact_response.json()["artifacts"]:
+        if artifact["name"].startswith(file_name):
+            download_url = artifact["archive_download_url"]
+            break
+
+    if not download_url:
+        get_console().print(f"[error]No artifact found for {file_name}")
+        sys.exit(1)
+
+    get_console().print(f"[info]Downloading artifact from {download_url} to {output_file}")
+
+    response = session.get(download_url, stream=True, headers=headers)
+
+    if response.status_code != 200:
+        get_console().print(
+            "[error] Downloading artifact failed with status code %s, "
+            "you might need to provide GITHUB_TOKEN, set it as environment variable",
+            response.status_code,
+        )
+        sys.exit(1)
+
+    total_size = int(response.headers.get("content-length", 0))
+    temp_file = tempfile.NamedTemporaryFile().name + "/file.zip"
+    os.makedirs(os.path.dirname(temp_file), exist_ok=True)
+
+    with tqdm(total=total_size, unit="B", unit_scale=True, desc=temp_file, ascii=True) as progress_bar:
+        with open(temp_file, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1 * 1024 * 1024):
+                if chunk:
+                    f.write(chunk)
+                    progress_bar.update(len(chunk))
+
+    with zipfile.ZipFile(temp_file, "r") as zip_ref:
+        zip_ref.extractall("/tmp/")
+
+    os.remove(temp_file)
+
+
+def download_artifact_from_pr(pr: str, output_file: str):
+    import requests
+
+    pr_url = f"https://api.github.com/repos/apache/airflow/pulls/{pr}"
```

Comment:
Nit: we should also support #PR form - similarly as `gh checkout pr`does

Reply:
Sorry didnt get :), this method is for pr from, it needs pr number. we can provide `--from-pr 123`

Reply:
You mean pull request number with hash? like `--from-pr #45287`?

Reply:
got it resolved :)